File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    var answer = 0;
    
    var window = 1;
    
    var leng = elements.length;
    var set = new Set();
    while(window <= leng){
        for(let i = 0;i<leng;i++){
            var sum =0;
            for(let j = i;j<i+window;j++){
                if(j>=leng){
                    sum+=elements[j-leng];
                }else{
                    sum+=elements[j];
                }
            }
            set.add(sum);
            sum =0;
        }
        window++;
    }
    answer = set.size;
    return answer;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <unordered_set>
#include <iostream>
using namespace std;

unordered_set<int> us;

int solution(vector<int> elements) {
    int answer = 0;
    
    int window =1;
    //슬라이딩 윈도우 (1~원소의 최대개수)
    
    while(window<=elements.size()){
        int left=0,right=window-1;
        int sum =0;
        
        for(int i = left;i<elements.size();i++){
            for(int j = i;j<i+window;j++){
                if(j >= elements.size()){
                    sum+=elements[j-elements.size()];
                }else{
                    sum+=elements[j];
                }
            }
            //cout << sum << " " << window << "\n";
            us.insert(sum);
            sum = 0;
        }
        window++;
    }
    answer = us.size();
    for(auto v: us){
        //cout << v << " ";
    }
    return answer;
}